runtime/src/stakes.rs

```diff
@@ -544,6 +547,23 @@ impl From<Stakes<StakeAccount>> for Stakes<Delegation> {
     }
 }
 
+impl<'a> From<&'a Stakes<StakeAccount>> for Stakes<&'a Delegation> {
+    fn from(stakes: &'a Stakes<StakeAccount>) -> Self {
```

Comment:
it looks like this is only used by serialize(), is that right? if so, I think we could
avoid this big allocation entirely and stream-serialize the input
`&Stakes<StakeAccount>` without creating this temporary `Stakes<&Delegation>`
directly in the caller?

Reply:
Yeah good call, let me know what you think of https://github.com/anza-xyz/agave/pull/2455/commits/36b5a16695c551eacdadca29d5a09282536457b3

Reply:
New commit here with generics and serialize-with removed: https://github.com/anza-xyz/agave/pull/2455/commits/e00471861436ea4ef060c7b51f9809603036d57f